zoloto/marker.py

```diff
@@ -33,85 +34,101 @@ def id(self) -> int:
-        return ThreeDCoordinates(*self._tvec)
-
-    @lru_cache(maxsize=None)
-    def _get_pose_vectors(self) -> Tuple[ndarray, ndarray]:
-        # Check if the Marker is eager.
-        # We cannot call _is_eager, else mypy will think we are returning Optional
-        if self.__precalculated_vectors is not None:
-            return self.__precalculated_vectors
-
-        if self.__camera_calibration_params is None:
-            raise MissingCalibrationsError()
-
-        rvec, tvec, _ = aruco.estimatePoseSingleMarkers(
-            [self.__pixel_corners],
-            self.__size,
-            self.__camera_calibration_params.camera_matrix,
-            self.__camera_calibration_params.distance_coefficients,
-        )
-        return rvec[0][0], tvec[0][0]
+        return ThreeDCoordinates(*self._tvec.tolist())
 
     @property
     def _rvec(self) -> ndarray:
-        rvec, _ = self._get_pose_vectors()
-        return rvec
+        return self._get_pose_vectors()[0]
 
     @property
     def _tvec(self) -> ndarray:
-        _, tvec = self._get_pose_vectors()
-        return tvec
+        return self._get_pose_vectors()[1]
+
+    def __json__(self) -> str:
+        return encode_as_json(self.as_dict())
 
     def as_dict(self) -> Dict[str, Any]:
         marker_dict = {
             "id": self.id,
             "size": self.size,
-            "pixel_corners": self.__pixel_corners.tolist(),  # type: ignore
+            "pixel_corners": [list(corner) for corner in self.pixel_corners],
         }
         try:
-            marker_dict.update({"rvec": list(self._rvec), "tvec": list(self._tvec)})
+            marker_dict.update(
+                {"rvec": self._rvec.tolist(), "tvec": self._tvec.tolist()}
+            )
         except MissingCalibrationsError:
             pass
         return marker_dict
 
-    @classmethod
-    def from_dict(cls, marker_dict: Dict[str, Any]) -> "Marker":
-        marker_args = [
-            marker_dict["id"],
-            array(marker_dict["pixel_corners"]),
-            marker_dict["size"],
-            None,
-        ]
-        if "rvec" in marker_dict and "tvec" in marker_dict:
-            marker_args.append((array(marker_dict["rvec"]), array(marker_dict["tvec"])))
-        return cls(*marker_args)
+
+class EagerMarker(BaseMarker):
+    def __init__(
+        self,
+        marker_id: int,
+        corners: List[ndarray],
+        size: int,
+        marker_dict: MarkerDict,
+        precalculated_vectors: Tuple[ndarray, ndarray],
+    ):
+        super().__init__(marker_id, corners, size, marker_dict)
+        self.__precalculated_vectors = precalculated_vectors
+
+    def _get_pose_vectors(self) -> Tuple[ndarray, ndarray]:
+        return self.__precalculated_vectors
+
+
+class Marker(BaseMarker):
+    def __init__(
+        self,
+        marker_id: int,
+        corners: List[ndarray],
+        size: int,
+        marker_dict: MarkerDict,
+        calibration_params: Optional[CalibrationParameters],
```

Comment:
Imo, `calibration_params` should explicitly be a `kwarg` here.

```suggestion
        *,
        calibration_params: Optional[CalibrationParameters],
```